src/prpy/util.py

```diff
@@ -497,6 +497,21 @@ def stop(self):
     def get_duration(self):
         return self.end - self.start
 
+class Watchdog(object):
+    def __init__(self, timeout_duration, handler):
+        self.timeout_duration = timeout_duration
+        self.handler = handler
+        self.timer = threading.Timer(self.timeout_duration, self.handler)
+        self.timer.start()
+
+    def reset(self):
+        self.timer.cancel()
+        self.timer = threading.Timer(self.timeout_duration, self.handler)
+        self.timer.start()
```

Comment:
I am a bit concerned about creating a new `Timer` every time we send a velocity command. My hunch is that every `Timer` starts a new thread. This will create and destroy a huge number of threads. There may be some gotchas about this, e.g. thread exhaustion, performance implications, but I am not positive. Maybe @cdellin or @psigen is more informed?

If this indeed an issue, then it is relatively straightforward to use one long-running thread per `Watchdog` by setting the expiration timestamp as a member variable.